$NetBSD: patch-ab,v 1.3 2007/05/23 11:13:42 rillig Exp $

The test for socklen_t is horribly broken. However, it worked on NetBSD,
but with a compiler warning.

--- configure.orig	2002-04-11 00:50:20.000000000 +0200
+++ configure	2007-05-23 13:08:25.000000000 +0200
@@ -876,16 +876,16 @@ if test "x$enable_fsstd" = "xyes"; then
 else
 	ICECAST_BINDIR=${prefix}/bin
 	ICECAST_SBINDIR=${prefix}/bin
-	ICECAST_ETCDIR=conf
-	ICECAST_ETCDIR_INST=${prefix}/conf
+	ICECAST_ETCDIR=${sysconfdir}
+	ICECAST_ETCDIR_INST=${prefix}/share/examples/icecast
 	ICECAST_SHAREDIR=${prefix}
 	ICECAST_LOGDIR=logs
-	ICECAST_LOGDIR_INST=${prefix}/logs
+	ICECAST_LOGDIR_INST=${VARBASE}/log/icecast
 	ICECAST_TEMPLATEDIR=templates
-	ICECAST_TEMPLATEDIR_INST=${prefix}/templates
+	ICECAST_TEMPLATEDIR_INST=${prefix}/share/icecast/templates
 	ICECAST_STATICDIR=static
-	ICECAST_STATICDIR_INST=${prefix}/static
-	ICECAST_DOCDIR=${prefix}/doc
+	ICECAST_STATICDIR_INST=${prefix}/share/icecast/static
+	ICECAST_DOCDIR=${prefix}/share/doc/icecast
 	USE_FSSTD=no
 fi
 
@@ -1540,7 +1540,7 @@ EOF
 
 		echo "$ac_t""BSD/OS" 1>&6
 		;;
-	*BSD*)
+	unused)
 		__SOMEBSD__=yes
 		cat >> confdefs.h <<\EOF
 #define __SOMEBSD__ 1
@@ -3459,10 +3459,9 @@ cat > conftest.$ac_ext <<EOF
 #include "confdefs.h"
 #include <sys/types.h>
 		#include <sys/socket.h>
+typedef socklen_t mytype; 
 int main() {
-typedef socklen_t int; 
-	    int apan (void) {return 1;}
-; return 0; }
+return 0; }
 EOF
 if { (eval echo configure:3468: \"$ac_compile\") 1>&5; (eval $ac_compile) 2>&5; }; then
   rm -rf conftest*
